zio-http/src/main/scala/zhttp/http/Proxy.scala

```diff
@@ -0,0 +1,37 @@
+package zhttp.http
+import io.netty.handler.codec.http.EmptyHttpHeaders
+import io.netty.handler.proxy.HttpProxyHandler
+import zhttp.http.middleware.Auth.Credentials
+
+import java.net.InetSocketAddress
+
+final case class Proxy(
+  url: URL,
+  credentials: Option[Credentials] = None,
+  headers: Option[Headers] = None,
+) {
+
+  /**
+   * Converts a Proxy to [io.netty.handler.proxy.HttpProxyHandler]
+   */
+  private[zhttp] def encode: Option[HttpProxyHandler] = credentials.fold(unauthorizedProxy)(authorizedProxy)
+
+  private def authorizedProxy(credentials: Credentials): Option[HttpProxyHandler] = for {
+    proxyAddress <- buildProxyAddress
+    uname          = credentials.uname
+    upassword      = credentials.upassword
+    encodedHeaders = headers.map(_.encode).getOrElse(EmptyHttpHeaders.INSTANCE)
+  } yield new HttpProxyHandler(proxyAddress, uname, upassword, encodedHeaders)
+
+  private def unauthorizedProxy: Option[HttpProxyHandler] = for {
+    proxyAddress <- buildProxyAddress
+    encodedHeaders = headers.map(_.encode).getOrElse(EmptyHttpHeaders.INSTANCE)
+  } yield {
+    new HttpProxyHandler(proxyAddress, encodedHeaders)
+  }
+
+  private def buildProxyAddress: Option[InetSocketAddress] = for {
+    proxyHost <- url.host
+    proxyPort <- url.port
+  } yield new InetSocketAddress(proxyHost, proxyPort)
+}
```

Comment:
We can also add some helper methods and use builder pattern to build proxy in parts. eg :

```scala
def withCredentials(cred: Credential): Proxy = ???
def withHeaders(headers: Headers): Proxy = ???
```